### PR TITLE
Add Dev versions with Emotion Variable Name Labels

### PR DESCRIPTION
**Pull request by azizhk**
Need to add labels so that it eases development and we see classNames to be equal to our variable names.
If we add babel-plugin-emotion in our consuming repos then because rollup adds `$1` suffix to duplicate variable names and as `$` is not allowed in CSS Classname Selector Syntax, the styles don't get applied. I've fixed this in `babel-plugin-emotion` but also created this alternative. 
https://github.com/emotion-js/emotion/pull/865

### CODE REVIEW

**Comment by ritz078** [comment]
After this change it seems it's no longer tree shakeable. run `yarn test:shaking`.

**Comment by azizhk** [comment]
```
error Command "test:shaking" not found.
```
Is something missing?

**Comment by ritz078** [comment]
run yarn once

**Comment by azizhk** [comment]
Nope still nothing.

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "test:cover": "npm run test && codecov",
     "test": "jest",
     "prepublishOnly": "npm run test && npm run build",
+    "test:shaking": "agadoo",
```
Comment (ritz078) [inline package.json]: ye to hai command.

**Comment by azizhk** [comment]
Will reopen when ready.

**Comment by azizhk** [comment]
Let me add tests for tree shake-ability using webpack as well.
See if that works and then let's go ahead with this.

**Comment by ritz078** [comment]
@azizhk any update on this ?

**Comment by azizhk** [comment]
Nope. This can be merged as is. Was thinking of adding webpack tree shaking tests as well.

**Comment by VivekNayyar** [comment]
The latest version of emotion above v8 does not require babel plug-in emotion and it's autoLabel takes care of removing the $1 suffix.

Do we still need babel plug-in emotion?

**Comment by azizhk** [comment]
Yup, I guess I should rephrase, that we dont "need" babel-plugin-emotion but its a "good to have" as it adds the variable names as part of the css classname which helps development by making it easier to find in the code through autoLabel

**Comment by VivekNayyar** [comment]
Understood. Fair point. I was under the assumption that emotion has taken care of that as well in the core package but went through the docs again and it seems babel plug-in emotion is needed for adding variable names as part of classname through autoLabel.

**Comment by azizhk** [comment]
Ok so I've ditched the [React like](https://github.com/facebook/react/blob/master/packages/react/npm/index.js) runtime based environment detection.
By default the production version will be provided through node_modules. User can use the development by specifying the `.dev.js` file.

**Pull request merged**